src/NuGet.Clients/NuGet.PackageManagement.UI/Options/AddMappingDialog.xaml

```diff
@@ -0,0 +1,133 @@
+      <RowDefinition Height="36"/>
+    </Grid.RowDefinitions>
+    <Grid.Resources>
+      <Style TargetType="ScrollBar">
+        <Setter Property="Background" Value="{DynamicResource {x:Static SystemColors.ScrollBarColorKey}}"/>
+      </Style>
+    </Grid.Resources>
+    <StackPanel Grid.Row="0">
+      <Grid>
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="*"/>
+          <ColumnDefinition Width="24"/>
+        </Grid.ColumnDefinitions>
+        <TextBlock AutomationProperties.Name="AddNewPackageNamespace"
+                   Grid.Column="0"
+                   Text="{x:Static nuget:Resources.VSOptions_Label_AddPackageNamespace}"
+                   Margin="12,12,0,0"/>
+        <Button x:Uid="ButtonClose"
+                       Grid.Column="1"
+                       HorizontalAlignment="Left"
+                       VerticalAlignment="Top"
+                       Margin="0,6,6,0"
+                       Command="{Binding HideDialogCommand}"
+                       AutomationProperties.Name="Close"
+                       IsCancel="True"
+                       BorderBrush="Transparent"
+                       BorderThickness="0"
+                       Background="{Binding Path=(Window.Background), RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}">
+          <Canvas x:Uid="canvas" Width="12" Height="10">
+            <Path x:Uid="CloseButtonPath"
+                  Width="10"
+                  Height="8"
+                  Fill="{Binding Path=(TextElement.Foreground), RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}"
+                  Stretch="Uniform"
+                  Data="F1 M 0,0L 2,0L 5,3L 8,0L 10,0L 6,4L 10,8L 8,8L 5,5L 2,8L 0,8L 4,4L 0,0 Z" />
+          </Canvas>
+        </Button>
+      </Grid>
+    </StackPanel>
+    <StackPanel Grid.Row="1">
+      <Grid>
+        <Grid.RowDefinitions>
+          <RowDefinition Height="48"/>
+          <RowDefinition Height="*"/>
+          <RowDefinition Height="36"/>
+        </Grid.RowDefinitions>
+        <vsui:WatermarkedTextBox Grid.Row="0"
+                                 AutomationProperties.Name="Package ID"
+                                 x:Name="_packageID"
+                                 MaxLength="100"
+                                 Margin="12,6,12,6"
+                                 Watermark="{x:Static nuget:Resources.VSOptions_Watermark_AddPackageNamespace}"
+                                 Foreground="{Binding Path=(TextElement.Foreground), RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}"
+                                 Background="{Binding Path=(Window.Background), RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}"
+                                 TextChanged="PackageID_TextChanged"/>
+        <StackPanel Orientation="Vertical"  Grid.Row="1">
+          <nuget:ProjectsListView AutomationProperties.Name="Sources List"
```

Comment:
The control I needed to reuse was unfortunately named specifically for Projects list, but I'll refactor that in a separate PR: https://github.com/NuGet/Client.Engineering/issues/1903